File: reto2/reto2.js
function createFrame(names) {
  let width = 0;

  // Obtener el ancho máximo
  for (const name of names) if (name.length > width) width = name.length;

  // Borde superior
  let frame = `${"*".repeat(width + 4)}\n`;

  // Contenido del frame
  for (const name of names)
    frame += `* ${name}${" ".repeat(width - name.length + 1)}*\n`;

  // Borde inferior
  frame += "*".repeat(width + 4);

  return frame;
}

File: reto2/reto2.py
def createFrame(names):
    width = 0

    for name in names:
        if len(name) > width:
            width = len(name)

    frame = "*" * (width + 4) + "\n"

    for name in names:
        padding = " " * (width - len(name) + 1)
        frame += f"* {name}{padding}*\n"

    frame += "*" * (width + 4)

    return frame
